### ISSUE TITLE
Redundant detected boxes

### ISSUE THREAD

**Issue by wuhanshuo** [THU-MIG/yolov10#236]
My code:
```python
from ultralytics import YOLOv10

model = YOLOv10("weights/trained_v10m.pt")
video_path = "my/video/path"

results = model(source=video_path, conf=0.25, device=[0], save=False, show=False, stream=True)
for idx, result in enumerate(results):
    dets = result.boxes.data.cpu().numpy() # --> Matrix (x, y, x, y, conf, cls)
    print("dets:", dets)
```

Outputs in frame 266:
```
video 1/1 (frame 266/7397) /cluster/my/path/video.mp4: 384x640 3 wheels, 1 car, 1 minivan, 10.3ms
dets: [[     1212.2      309.88      1464.7      482.17     0.98511           2]
 [     1290.3      425.93      1318.7      480.26     0.95292           0]
 [     721.58       360.8      767.41      394.72     0.93868           1]
 [     1213.4      413.81        1234      461.19     0.50953           0]
 [     1213.1       412.4      1236.8      460.92     0.29856           0]]
```

The last two boxes are almost the same one but detected twice. Why does YOLOv10 have this issue by replacing NMS with consistent dual assignments?

**Comment by lryan599** [comment]
I also have the same problem.

**Comment by jameslahm** [comment]
Thanks for your interest! Do you use the finetuned model to perform predictions? Could you please try to train the model for more epochs? Thanks!

**Comment by wuhanshuo** [comment]
> Thanks for your interest! Do you use the finetuned model to perform predictions? Could you please try to train the model for more epochs? Thanks!

@jameslahm Many thanks for your comments. I trained the model on my dataset starting from your pretrained weights. No matter how I change epochs from 900~2000 and other training parameters, all models keep giving redundant detection boxes.

One of trainings:
![image](https://github.com/THU-MIG/yolov10/assets/63944310/dfc6da6d-5c6f-4d1e-afed-0b35de112e8f)

I also submit a issue [here](https://github.com/ultralytics/ultralytics/issues/13894#issuecomment-2184976714). Now the problem is solved by adding an NMS as prediction post-process...

**Comment by jameslahm** [comment]
@wuhanshuo Thanks! What if you set a higher confidence threshold than that of the redundant prediction with the lower score?

**Comment by lryan599** [comment]
> @wuhanshuo Thanks! What if you set a higher confidence threshold than that of the redundant prediction with the lower score?

But I think it is hard to set an appropriate confidence threshold...

After all it is unpredictable...

**Comment by wuhanshuo** [comment]
> @wuhanshuo Thanks! What if you set a higher confidence threshold than that of the redundant prediction with the lower score?

@jameslahm Yes I believe a higher confidence will remove redundant boxes significantly. However I will not consider it as the first option, since in my task a higher confidence may miss some small objects...
As @lryan599 mentioned, it is unpredictable and may harm the generalization ability of the model in difference scenes... Also it is hard to find a perfect confidence.

**Comment by lryan599** [comment]
@wuhanshuo Thank you for your suggestion! I will give it a try. @jameslahm Maybe you can add NMS as an optional argument? After all, in a single prediction, there are only a few duplicate detection boxes, so the NMS will not take too much time, I think. This is for engineering considerations.

**Comment by jameslahm** [comment]
@lryan599 Thanks for your suggestion! That is a good solution and we will consider this.

**Comment by wuhanshuo** [comment]
> @wuhanshuo Thanks! Would you mind sharing your example, including the checkpoint and the detected image with us? We will also try to find other solutions.

@jameslahm Due to project confidentiality reasons, I cannot post my data here. I have sent an email to your gmail address with my checkpoint and an example image.

**Comment by jameslahm** [comment]
@wuhanshuo Thanks a lot! We have received your email.
